### PR TITLE
Fix flaky test (PooledClientProviderTest)

### PR DESCRIPTION
**Pull request by anwar6953**
Fix flaky test (PooledClientProviderTest). Previously, the entire client pool was not exhausted.

### CODE REVIEW

File: tephra-core/src/test/java/org/apache/tephra/distributed/PooledClientProviderTest.java [hunk 1]
```diff
@@ -137,15 +137,15 @@ public Boolean call() throws Exception {
 
 
     //Now race to get MAX_CLIENT_COUNT+1 clients, exhausting the pool and requesting 1 more.
-    List<Future<Integer>> clientIds = new ArrayList<Future<Integer>>();
-    CountDownLatch countDownLatch = new CountDownLatch(1);
+    List<Future<Integer>> clientIds = new ArrayList<>();
+    // we want to ensure that all clients have been exhausted before releasing any
+    CountDownLatch countDownLatch = new CountDownLatch(MAX_CLIENT_COUNT);
```
Comment (poornachandra) [inline tephra-core/src/test/java/org/apache/tephra/distributed/PooledClientProviderTest.java]: Would be good to call this `clientDoneLatch`

File: tephra-core/src/test/java/org/apache/tephra/distributed/PooledClientProviderTest.java [hunk 2]
```diff
@@ -154,11 +154,10 @@ public Boolean call() throws Exception {
     // now, try it again with, where each thread holds onto the client for twice the client.obtain.timeout value.
```
Comment (poornachandra) [inline tephra-core/src/test/java/org/apache/tephra/distributed/PooledClientProviderTest.java]: Comments are outdated now

File: tephra-core/src/test/java/org/apache/tephra/distributed/PooledClientProviderTest.java (hunk 1 above)
Comment (anwar6953) [inline tephra-core/src/test/java/org/apache/tephra/distributed/PooledClientProviderTest.java]: done

**Comment by poornachandra** [comment]
LGTM

**Pull request closed without merging**